### PR TITLE
Add info/serverinfo command

### PR DESCRIPTION
**Pull request by wasdennnoch**
**Please describe the changes this PR makes and why it should be merged:**

**What does this PR do:**
- [x] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes command functions
  - [x] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (endpoint functions)
  - [ ] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

### CODE REVIEW

File: Commands/Public/info.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		return ":flag_eu:";
+	}
+	if (region.startsWith("us-")) {
+		return ":flag_us:";
+	}
+	return ":interrobang:";
+};
+const getDescriptionForVerificationLevel = level => {
+	switch (level) {
+		case 0:
```
Comment (wasdennnoch) [inline Commands/Public/info.js]: Actually, this could be replaced by an array I guess?

File: Commands/Public/info.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+const moment = require("moment");
```
Comment (vladfrangu) [inline Commands/Public/info.js]: Can you use moment-timezone + `.tz("Europe/London")` to be UTC?

File: Commands/Public/info.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	if (region.startsWith("eu-")) {
+		return ":flag_eu:";
+	}
+	if (region.startsWith("us-")) {
+		return ":flag_us:";
+	}
+	return ":interrobang:";
+};
+const getDescriptionForVerificationLevel = level => {
```
Comment (vladfrangu) [inline Commands/Public/info.js]: This can be under constants.

File: Commands/Public/info.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
+const moment = require("moment");
+
+const getFlagForRegion = region => {
```
Comment (vladfrangu) [inline Commands/Public/info.js]: Could be a util

File: Commands/Public/info.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+			return "High - (╯°□°）╯︵ ┻━┻ - must be a member of the server for longer than 10 minutes";
+		case 4:
+			return "Very High - ┻━┻ミヽ(ಠ益ಠ)ﾉ彡┻━┻ - must have a verified phone number";
+	}
+};
+
+module.exports = async ({ client, Constants: { Colors } }, { serverDocument }, msg, commandData) => {
+	let commandUses = 0;
+	for (const k in serverDocument.command_usage) {
+		commandUses += serverDocument.command_usage[k];
```
Comment (vladfrangu) [inline Commands/Public/info.js]: `let commandUses = Object.values(serverDocument.command_usage).reduce((a, b) => a + b, 0);`

File: Commands/Public/info.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	const guild = msg.guild;
+	const created = moment(guild.createdTimestamp);
+	const onlineMembers = guild.members.filter(m => m.presence.status === "online").size;
+	const region = guild.region;
+	const regionInfo = (await guild.fetchVoiceRegions()).get(region);
+	const publicData = serverDocument.config.public_data;
+
+	const embedFields = [];
+	const generalText = [
+		`**»** Guild ID: ${guild.id}`,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: This can go in the title (`${guildname} :: ${guildid}`)

File: Commands/Public/info.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+		`**»** Owner: ${guild.owner}`,
+		`**»** Created: ${created.format("DD-MM-YYYY HH:mm:ss")} (${created.fromNow()})`,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: add an `[at]` in the created date

File: Commands/Public/info.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+		`**»** Voice region: ${regionInfo ? regionInfo.name : region || "Unknown"} ${getFlagForRegion(region)}${regionInfo && regionInfo.deprecated ? " (DEPRECATED)" : ""}`,
+		`**»** Verification level: ${getDescriptionForVerificationLevel(guild.verificationLevel)}`,
+	];
+	if (!configJSON.activityBlocklist.includes(guild.id) && publicData.isShown) {
+		generalText.push(`**»** This server is shown on the activity page using the category '${publicData.server_listing.category}'${publicData.server_listing.isEnabled ? ". Everyone can join it from there" : ""}`);
+	}
+	embedFields.push({
+		name: "General Info :pencil:",
+		value: generalText.join("\n"),
+		inlinee: false,
```
Comment (wasdennnoch) [inline Commands/Public/info.js]: typo'd

File: Commands/Public/info.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+	const created = moment(guild.createdTimestamp);
+	const onlineMembers = guild.members.filter(m => m.presence.status === "online").size;
+	const region = guild.region;
+	const regionInfo = (await guild.fetchVoiceRegions()).get(region);
+	const publicData = serverDocument.config.public_data;
+
+	const embedFields = [];
+	const generalText = [
+		`**»** Guild ID: ${guild.id}`,
+		`**»** Owner: ${guild.owner}`,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: How would the owner look under the author field?

File: Commands/Public/info.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+		`**»** Created: ${created.format("DD-MM-YYYY HH:mm:ss")} (${created.fromNow()})`,
+		`**»** Voice region: ${regionInfo ? regionInfo.name : region || "Unknown"} ${getFlagForRegion(region)}${regionInfo && regionInfo.deprecated ? " (DEPRECATED)" : ""}`,
+		`**»** Verification level: ${getDescriptionForVerificationLevel(guild.verificationLevel)}`,
+	];
+	if (!configJSON.activityBlocklist.includes(guild.id) && publicData.isShown) {
+		generalText.push(`**»** This server is shown on the activity page using the category '${publicData.server_listing.category}'${publicData.server_listing.isEnabled ? ". Everyone can join it from there" : ""}`);
+	}
+	embedFields.push({
+		name: "General Info :pencil:",
```
Comment (vladfrangu) [inline Commands/Public/info.js]: Use the unicode please

File: Commands/Public/info.js [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+		value: generalText.join("\n"),
+		inlinee: false,
+	});
+	const numbersText = [
+		`**»** Members: ${guild.memberCount} (of which ${onlineMembers} ${onlineMembers === 1 ? "is" : "are"} online)`,
+		`**»** Text channels: ${guild.channels.filter(c => c.type === "text").size}`,
+		`**»** Voice channels: ${guild.channels.filter(c => c.type === "voice").size}`,
+		`**»** Channel Categories: ${guild.channels.filter(c => c.type === "category").size}`,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: c o n s i s t e n t  c a p i t a l i z a t i o n

File: Commands/Public/info.js [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+		`**»** Roles: ${guild.roles.size - 1}`,
+		`**»** Custom emoji: ${guild.emojis.size}`,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: `Custom emoji${guild.emojis.size === 1 ? "" : "s"}`

File: Commands/Public/info.js [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+		`**»** Messages today: ${serverDocument.messages_today}`,
+		`**»** Commands used this week: ${commandUses}`,
+	];
+	embedFields.push({
+		name: "Crunchy Numbers :1234:",
```
Comment (vladfrangu) [inline Commands/Public/info.js]: Unicode!!

File: Commands/Public/info.js [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	if (guild.features.includes("INVITE_SPLASH")) {
+		specialText.push(`**»** This server can use a **custom invite splash background**!${guild.splash ? ` It is currently set to [this](${guild.splashURL({ format: "png", size: 2048 })})` : ""}`);
+	}
+	if (guild.features.includes("VANITY_URL")) {
+		let customInvite;
+		if (guild.me.permissions.has("MANAGE_GUILD")) {
+			customInvite = (await client.api.guilds(guild.id)["vanity-url"].get()).code;
+		}
+		specialText.push(`**»** This server can use a **custom vanity URL**!${customInvite ? ` It is currently set to https://discord.gg/${customInvite}` : ""}`);
```
Comment (vladfrangu) [inline Commands/Public/info.js:80]: Maybe make it say that we can't find out the code? /shrug

File: Commands/Public/info.js [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	if (specialText.length) {
+		embedFields.push({
+			name: "Special Features :star:",
```
Comment (vladfrangu) [inline Commands/Public/info.js]: You know what I'm about to say

File: Commands/Public/info.js [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+			value: specialText.join("\n"),
+			inline: false,
+		});
+	}
+
+	msg.send({
+		embed: {
+			color: Colors.LIGHT_GREEN,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: Should probably be Colors.INFO

File: Commands/Public/info.js [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+			title: `Information for ${guild.name}`,
+			fields: embedFields,
```
Comment (vladfrangu) [inline Commands/Public/info.js]: could rename embedFields to fields, then use the magic that is `fields,`

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Sorry babe!

File: Commands/Public/info.js (hunk 9 above)
Comment (wasdennnoch) [reply Commands/Public/info.js]: Wouldn't be immediately clear that it's the owner, at least for me. Plus it's not an embedded mention :P

**Pull request merged**